components/TreasuryAccount/HoldTokensTotalPrice.tsx

```diff
@@ -0,0 +1,48 @@
+import { getMintMetadata } from '@components/instructions/programs/splToken'
+import useInstructions from '@hooks/useInstructions'
+import {
+  formatMintNaturalAmountAsDecimal,
+  numberWithCommas,
+} from '@tools/sdk/units'
+import tokenService from '@utils/services/token'
+import { useEffect, useState } from 'react'
+
+const HoldTokensTotalPrice = () => {
+  const { governedTokenAccounts } = useInstructions()
+  const [totalPrice, setTotalPrice] = useState('')
+  useEffect(() => {
+    async function calcTotalTokensPrice() {
+      const totalPrice = governedTokenAccounts
+        .filter(
+          (x) => typeof x.mint !== 'undefined' && typeof x.token !== 'undefined'
+        )
+        .map((x) => {
+          return (
+            parseFloat(
+              formatMintNaturalAmountAsDecimal(
+                x.mint!.account,
+                x.token!.account.amount
+              )
+                .split(',')
+                .join('')
+            ) *
+            tokenService.getUSDTokenPrice(
+              getMintMetadata(x.token!.account.mint)?.name
```

Comment:
Please use the mint address as the lookup and not the locally hardcoded symbol name